crypto/rand/rand_lib.c

```diff
@@ -469,12 +469,12 @@ static EVP_RAND_CTX *rand_new_seed(OSSL_LIB_CTX *libctx)
     EVP_RAND_CTX *ctx;
     char *name;
 
+    ERR_set_mark();
     name = dgbl->seed_name != NULL ? dgbl->seed_name : "SEED-SRC";
     rand = EVP_RAND_fetch(libctx, name, dgbl->seed_propq);
-    if (rand == NULL) {
-        ERR_raise(ERR_LIB_RAND, RAND_R_UNABLE_TO_FETCH_DRBG);
+    ERR_pop_to_mark();
+    if (rand == NULL)
```

Comment:
This seems hugely misplaced and inconsistent, especially since this function records proper errors for other error cases.
However, the only call of this function, in `RAND_get0_primary()`, doesn't check if the result is NULL or not, so it seems to me like the *sane* change would be something like this:

``` diff
diff --git a/crypto/rand/rand_lib.c b/crypto/rand/rand_lib.c
index 2ad3cf776f..f0284aab08 100644
--- a/crypto/rand/rand_lib.c
+++ b/crypto/rand/rand_lib.c
@@ -561,8 +561,11 @@ EVP_RAND_CTX *RAND_get0_primary(OSSL_LIB_CTX *ctx)
         if (!CRYPTO_THREAD_write_lock(dgbl->lock))
             return NULL;
 #ifndef FIPS_MODULE
-        if (dgbl->seed == NULL)
+        if (dgbl->seed == NULL) {
+            ERR_set_mark();
             dgbl->seed = rand_new_seed(ctx);
+            ERR_pop_to_mark();
+        }
 #endif
         if (dgbl->primary == NULL)
             dgbl->primary = rand_new_drbg(ctx, dgbl->seed,
```

Reply:
Okay, I've switched to this.